build-tools/packages/build-cli/src/codeCoverageAnalysisTool/getCommentForCodeCoverage.ts

```diff
@@ -0,0 +1,159 @@
+		);
+	}
+
+	if (newPackagesIdentifiedByCodeCoverage.length > 0) {
+		coverageSummaryForNewPackages = getCodeCoverageSummary(
+			newPackagesIdentifiedByCodeCoverage,
+		);
+	}
+	return {
+		commentMessage: [
+			title,
+			coverageSummaryForImpactedPackages,
+			coverageSummaryForNewPackages,
+			getSummaryFooter(baselineBuildInfo),
+		].join("\n\n"),
+		failBuild,
+	};
+};
+
+const getSummaryFooter = (baselineBuildInfo: IBaselineBuildMetrics): string => {
+	return `<hr><p>Baseline commit: ${
+		baselineBuildInfo.baselineCommit
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+	} <br>Baseline build: <a target="_blank" href=${baselineBuildInfo.baselineBuild._links?.web.href as string}> ${
+		baselineBuildInfo.baselineBuild.id
+	} </a><br> Happy Coding!!</p>`;
+};
+
+const getCodeCoverageSummary = (
+	codeCoverageComparisonReport: CodeCoverageComparison[],
+): string => {
+	const summary = codeCoverageComparisonReport
+		.sort((report1, report2) => report1.branchCoverageDiff - report2.branchCoverageDiff)
+		.map((coverageReport) => getCodeCoverageSummaryForPackages(coverageReport))
+		.reduce((prev, current) => prev + current);
+
+	return summary;
+};
+
+const getCodeCoverageSummaryForPackages = (coverageReport: CodeCoverageComparison): string => {
+	const metrics = codeCoverageDetailsHeader + getMetricRows(coverageReport);
+
+	return `<details><summary><b>${getColorGlyph(coverageReport.branchCoverageDiff)} ${
+		coverageReport.packageName
+	}:</b> ${formatDiff(coverageReport.branchCoverageDiff)}</summary>${metrics}</table></details>`;
+};
+
+const getColorGlyph = (codeCoverageBranchDiff: number): string => {
+	if (codeCoverageBranchDiff === 0) {
+		return '<span style="color: green">■</span>';
+	}
+
+	if (codeCoverageBranchDiff > 0) {
+		return '<span style="color: green">⯅</span>';
+	}
+
+	return '<span style="color: red">⯆</span>';
+};
+
+const formatDiff = (branchCoverageDiff: number): string => {
+	if (branchCoverageDiff === 0) {
+		return "No change";
+	}
+	return `${(branchCoverageDiff * 100).toFixed(2)}%`;
+};
+
+const getMetricRows = (codeCoverageComparisonReport: CodeCoverageComparison): string => {
+	const glyphForLineCoverage = getColorGlyph(codeCoverageComparisonReport.lineCoverageDiff);
+	const glyphForBranchCoverage = getColorGlyph(
+		codeCoverageComparisonReport.branchCoverageDiff,
+	);
+
+	return (
+		`<tr>
+    <td>Branch Coverage</td>
+    <td>${formatDiff(codeCoverageComparisonReport.branchCoverageInBaseline)}</td>
+    <td>${formatDiff(codeCoverageComparisonReport.branchCoverageInPr)}</td>
+    <td>${glyphForBranchCoverage} ${formatDiff(codeCoverageComparisonReport.branchCoverageDiff)}</td>
+  </tr>` +
+		`<tr>
+    <td>Line Coverage</td>
+    <td>${formatDiff(codeCoverageComparisonReport.lineCoverageInBaseline)}</td>
+    <td>${formatDiff(codeCoverageComparisonReport.lineCoverageInPr)}</td>
+    <td>${glyphForLineCoverage} ${formatDiff(codeCoverageComparisonReport.lineCoverageDiff)}</td>
```

Comment:
Looks big but it's just adding percent signs at the end of each cell
```suggestion
    <td>Branch Coverage</td>
    <td>${formatDiff(codeCoverageComparisonReport.branchCoverageInBaseline)}%</td>
    <td>${formatDiff(codeCoverageComparisonReport.branchCoverageInPr)}%</td>
    <td>${glyphForBranchCoverage} ${formatDiff(codeCoverageComparisonReport.branchCoverageDiff)}%</td>
  </tr>` +
		`<tr>
    <td>Line Coverage</td>
    <td>${formatDiff(codeCoverageComparisonReport.lineCoverageInBaseline)}%</td>
    <td>${formatDiff(codeCoverageComparisonReport.lineCoverageInPr)}%</td>
    <td>${glyphForLineCoverage} ${formatDiff(codeCoverageComparisonReport.lineCoverageDiff)}%</td>
```